src/Provider/EntityMutationMetadataProvider.php

```diff
@@ -185,6 +187,85 @@ public function isEntityManaged(EntityManagerInterface $em, $entity)
      */
     public function getFullChangeSet(EntityManagerInterface $em)
     {
-        return $em->getUnitOfWork()->getIdentityMap();
+        $change_set = [];
+
+        $managed    = $em->getUnitOfWork()->getIdentityMap();
+        $new        = $em->getUnitOfWork()->getScheduledEntityInsertions();
+
+        foreach ($managed as $class => $entities) {
+            $metadata = $em->getClassMetadata($class);
+
+            $change_set[$class] = $entities;
+            foreach ($entities as $entity) {
+                foreach ($this->checkAssociations($em, $metadata, $entity) as list($metadata_child, $child)) {
```

Comment:
A code clone of these lines is also just below. Can we think of some way to share this code?